load/TranscodeVideo.py

```diff
@@ -1,139 +1,146 @@
-			for jobId in self.openJobs:
-				response = self.client.read_job(Id=jobId).get("Job")
-				inputObj = response["Input"]["Key"]
-				status = response.get("Status")
-				if status == "Complete":
-					print("Job Complete", inputObj)
-
-				elif status == "Error":
-					output = response.get("Output", {})
-					errorMsg = output.get("StatusDetail", "")
-					if not errorMsg.startswith("3002"):
-						errorCount += 1
-					for output in response.get("Outputs", []):
-						if output.get("Status") == "Error":
-							print("ERROR %s" % (output.get("StatusDetail")))
-						else:
-							print("NO Message", output.get("Key"), output.get("StatusDetail"))
-
-				else:
-					stillOpenJobs.append(jobId)
-			self.openJobs = stillOpenJobs
-			time.sleep(10)
-		return errorCount == 0	
-
-
-if (__name__ == '__main__'):
-	from LanguageReaderCreator import LanguageReaderCreator	
-	from InputProcessor import *
-
-	config = Config.shared()
-	languageReader = LanguageReaderCreator("B").create(config.filename_lpts_xml)
-	# filesets = InputFileset.filesetCommandLineParser(config, languageReader)
-	filesets = InputProcessor.commandLineProcessor(config, AWSSession.shared().s3Client, languageReader)
-
-	for inp in filesets:
-		if inp.typeCode == "video":
-			TranscodeVideo.transcodeVideoFileset(config, inp.filesetPrefix, inp.s3FileKeys())
-
-# Successful tests with source on local drive
-# time python3 load/S3Utility.py test-video /Volumes/FCBH/all-dbp-etl-test/ ENGESVP2DV
-
-# Successful tests with source on s3
-
-
-
+# TranscodeVideo.py
+
+import json
+import boto3
+from S3Utility import *
+from RunStatus import *
+from AWSSession import *
+
+class TranscodeVideo:
+
+
+	def transcodeVideoFileset(config, filesetPrefix, s3FileKeys):
+		transcoder = TranscodeVideo(config, filesetPrefix)
+		RunStatus.printDuration("BEGIN SUBMIT TRANSCODE")
+		for s3FileKey in s3FileKeys:
+			transcoder.createJob(s3FileKey)
+		RunStatus.printDuration("BEGIN CHECK TRANSCODE")
+		done = transcoder.completeJobs()
+		if done:
+			print("Transcode %s succeeded." % (filesetPrefix))
+		else:
+			print("Transcode %s FAILED." % (filesetPrefix))
+		RunStatus.printDuration("TRANSCODE DONE")
+
+
+	## Used by UpdateDBPVideoTable
+	def getHLSTypes():
+		return ["_stream", "_av720p", "_av480p", "_av360p"]
+
+
+	def __init__(self, config, filesetPrefix):
+		self.config = config
+		self.filesetPrefix = filesetPrefix
+		self.videoPipeline = config.video_transcoder_pipeline
+		self.hls_audio_video_1080p = config.video_preset_hls_1080p
+		self.hls_audio_video_720p = config.video_preset_hls_720p
+		self.hls_audio_video_480p = config.video_preset_hls_480p
+		self.hls_audio_video_360p = config.video_preset_hls_360p
+		self.web_audio_video = config.video_preset_web
+		self.web_audio_video = config.video_preset_web
+		self.video_transcoder_mock = json.loads(config.video_transcoder_mock) if hasattr(config, 'video_transcoder_mock') else {}
+		if self.video_transcoder_mock.get("enable") == 1:
```

Comment:
it appears this will throw an error if self.video_transcoder_mock is null. may need null check
maybe one if-else block with the condition being hasattr(config, 'video_transcoder_mock') to include the code on lines 41-49. 
better yet, a method call to retrieve video transcoder which can be either the mock or the AWS resource

Reply:
@bradflood, I have updated this part to make sure that we will handle the error if video_transcoder_mock is null. Also, I have created a new method named getTranscoderClient to retrieve the video transcoder client depends on video_transcoder_mock flag. You can see it in the following commit:

https://github.com/faithcomesbyhearing/dbp-etl/pull/112/commits/f8e4425441d7a0dfd8cf9eff2418cf0443b752ad